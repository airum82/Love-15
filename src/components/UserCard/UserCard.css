.user-card {
  background-color: rgba(229, 149, 0, 1);
  width: 60%;
  margin: 0 auto;
}
