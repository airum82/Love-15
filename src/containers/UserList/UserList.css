.user-list {
  margin-top: 3%;
}