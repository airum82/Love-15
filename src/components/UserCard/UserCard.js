import React from 'react';
import './UserCard.css';
import PropTypes from 'prop-types';

export const UserCard = ({ username }) => {
  return (
    <div className="user-card">
      <h4>{username}</h4>
    </div>
  )
}

UserCard.propTypes = {
  username: PropTypes.string
}