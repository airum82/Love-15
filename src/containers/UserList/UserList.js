import React, { Component } from 'react';
import { UserCard } from '../../components/UserCard/UserCard';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class UserList extends Component {
  
  makeUserCard = () => {
    return this.props.userList.map((user, index) => {
      return (
        <UserCard
         key={index}
         username={user.username}
        />
      )
    })
  }

  render() {
    return(
      <div className="user-list">
        {this.props.userList.length ?
          this.makeUserCard() : ''}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  userList: state.userList
})

export default connect(mapStateToProps)(UserList);

UserList.propTypes = {
  userList: PropTypes.array
}