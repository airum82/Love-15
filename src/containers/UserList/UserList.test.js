import { UserList, mapStateToProps } from './UserList';
import { shallow } from 'enzyme';
import { UserCard } from '../../components/UserCard/UserCard';
import React from 'react';

describe('UserList', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <UserList
       userList={[{username: 'Bill'}, {username: 'Paulene'}]}
       />);
  })

  it('should match snapshot upon render with users', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot upon render with no users', () => {
    wrapper = shallow(
      <UserList
       userList={[]}
      />);
    expect(wrapper).toMatchSnapshot();
  })

  it('UserList should return an array of user cards', () => {
    const expectedOutput = [
      <UserCard
       key={0}
       username={'Bill'}
      />,
      <UserCard
       key={1}
       username={'Paulene'}
      />
    ];
    const result = wrapper.instance().makeUserCard();
    expect(result).toEqual(expectedOutput);
  })

  it('mapStateToProps should return an object with correct keys', () => {
    const mockState = {
      userList: [{}, {}]
    }
    const expectedOutput = {
      userList: [{}, {}]
    }
    const result = mapStateToProps(mockState);
    expect(result).toEqual(expectedOutput);
  })
})