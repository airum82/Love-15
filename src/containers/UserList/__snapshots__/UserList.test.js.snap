// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList should match snapshot upon render with no users 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserList
    userList={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "className": "user-list",
    },
    "ref": null,
    "rendered": "",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "user-list",
      },
      "ref": null,
      "rendered": "",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`UserList should match snapshot upon render with users 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserList
    userList={
        Array [
            Object {
              "username": "Bill",
            },
            Object {
              "username": "Paulene",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          username="Bill"
/>,
        <Unknown
          username="Paulene"
/>,
      ],
      "className": "user-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "username": "Bill",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "username": "Paulene",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            username="Bill"
/>,
          <Unknown
            username="Paulene"
/>,
        ],
        "className": "user-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "username": "Bill",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "username": "Paulene",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
